src/modbus_proxy.py

```diff
@@ -171,13 +172,30 @@ async def _write_read(self, data):
         await self._write(data)
         return await self._read()
 
+    def _transform_request(self, request):
+        request_unit_id = request[6]
+
+        if self.unit_id_remapping is None:
+            return request
+
+        if self.unit_id_remapping[request_unit_id] is None:
+            return request
+
+        remapped_unit_id = self.unit_id_remapping[request_unit_id]
+
+        self.log.info("remapping unit ID %s to %s", request_unit_id, remapped_unit_id)
+
+        request = request[:6] + bytearray([remapped_unit_id]) + request[7:]
+
+        return request
```

Comment:
if you don't mind, I'd like to propose an alternative, imho, simpler implementation:
```
    def _transform_request(self, request):
        uid = request[6]
        new_uid = self.unit_id_remapping.setdefault(uid, uid)
        if uid != new_uid:
            request = bytearray(request)
            request[6] = new_uid
            self.log.debug("remapping unit ID %s to %s", uid, new_uid)
        return request
```

assuming `self.unit_id_remapping` has been initialized with:

```self.unit_id_remapping = config.get("unit_id_remapping", {})```

Reply:
looks great!